test/test-helpers/create-entities.js

```diff
@@ -263,20 +263,32 @@ export async function createEdition(optionalBBID) {
 	return edition;
 }
 
-export async function createWork(optionalBBID) {
+export async function createWork(optionalBBID, optionalWorkAttribs) {
 	const bbid = optionalBBID || uuidv4();
 	await new Entity({bbid, type: 'Work'})
 		.save(null, {method: 'insert'});
 	await createEntityPrerequisites(bbid, 'Work');
-	const languageSetId = await createLanguageSet();
 
-	const workAttribs = {
-		bbid,
-		languageSetId,
-		typeId: random.number()
-	};
-	await new WorkType({id: workAttribs.typeId, label: `Work Type ${workAttribs.typeId}`})
-		.save(null, {method: 'insert'});
+	let workAttribs;
+	if (optionalWorkAttribs) {
+		workAttribs = optionalWorkAttribs;
+	}
+	else {
+		const languageSetId = await createLanguageSet();
+		workAttribs = {
+			bbid,
+			languageSetId,
+			typeId: random.number()
+		};
+	}
+	const workType = await new WorkType({id: workAttribs.typeId, label: `Work Type ${workAttribs.typeId}`})
+		.fetch({
+			require: false
+		});
+	if (!workType) {
+		await new WorkType({id: workAttribs.typeId, label: `Work Type ${workAttribs.typeId}`})
+			.save(null, {method: 'insert'});
+	}
```

Comment:
What do you think of this @MonkeyDo ? Bookshelf don't really have an upsert option. Do you know any better way to get this done?

Reply:
Unfortunately not :/